Spigot-Server-Patches/0181-Optimise-recalcBlockCounts.patch

```diff
@@ -0,0 +1,98 @@
+From 22c599cb8b0ec43840ed96715e4ceb501a16414e Mon Sep 17 00:00:00 2001
+From: Alfie Cleveland <alfeh@me.com>
+Date: Wed, 23 Nov 2016 19:03:00 +0000
+Subject: [PATCH] Optimise recalcBlockCounts
+
+
+diff --git a/src/main/java/net/minecraft/server/ChunkSection.java b/src/main/java/net/minecraft/server/ChunkSection.java
+index afdc4a7..a390de3 100644
+--- a/src/main/java/net/minecraft/server/ChunkSection.java
++++ b/src/main/java/net/minecraft/server/ChunkSection.java
+@@ -45,20 +45,23 @@ public class ChunkSection {
+         IBlockData iblockdata1 = this.getType(i, j, k);
+         Block block = iblockdata1.getBlock();
+         Block block1 = iblockdata.getBlock();
+-
+-        if (block != Blocks.AIR) {
+-            --this.nonEmptyBlockCount;
+-            if (block.isTicking()) {
+-                --this.tickingBlockCount;
++        // Paper start
++        if (block != block1) {
```

Comment:
Based on what this change is doing, please undo the indentation change of all of this inner code.

I'm almost willing to say just short circuit return and copy the last line that isn't part of the if(), but that would introduce risk for bugs in future.

But im more comfortable with the inner code changing. and being ok still.